spec/function.dd

```diff
@@ -2181,15 +2181,50 @@ void main()
 $(H2 $(LNAME2 interpretation, Compile Time Function Execution (CTFE)))
 
     $(P Functions which are both portable and free of global side-effects can be
-    executed at compile time. This is useful when constant folding
-    algorithms need to include recursion and looping. Compile time function
-    execution is subject to the following restrictions:
+    executed at compile time. In certain contexts, such compile time execution
+    is guaranteed. It is called Compile Time Function Execution (CTFE) then.
+    The contexts that trigger CTFE are:)
+
+    $(UL
+    $(LI initialization of a static variable or a
+        $(DDSUBLINK spec/enum, manifest_constants, manifest constant))
+    $(LI static initializers of struct/class members)
+    $(LI dimension of a $(DDSUBLINK spec/arrays, static-arrays, static array))
+    $(LI argument for a $(DDSUBLINK spec/template, template_value_parameter,
+        template value parameter))
+    $(LI $(DDSUBLINK spec/version, staticif, $(D_KEYWORD static if)))
+    $(LI $(DDSUBLINK spec/version, staticforeach, $(D_KEYWORD static foreach)))
+    $(LI $(DDSUBLINK spec/version, static-assert, $(D_KEYWORD static assert)))
+    $(LI $(DDSUBLINK spec/statement, mixin-statement,
```

Comment:
These two cause a weird colouration on the link. Probably because one word is a keyword and the other is not. Not sure if it's fixable.

![image](https://user-images.githubusercontent.com/4450317/35295549-c586a93a-004f-11e8-8f3a-315815cb1a2d.png)

Reply:
Yep, it seems like it would be better to remove "statement" from the `mixin` and `pragma` list items.

Reply:
Perhaps @andralex can lend his typographical expertise?

Reply:
I've changed it to backticks instead of `$(D_KEYWORD ...)`. No more blue:

![spectacle sm5509](https://user-images.githubusercontent.com/9287500/35338683-d90faebe-011e-11e8-8fca-e3480564a24c.png)